src/main/java/de/tum/in/www1/artemis/service/EntityTitleCacheService.java

```diff
@@ -0,0 +1,125 @@
+        return titleMap.computeIfAbsent("course" + courseID, ignored -> courseRepository.getCourseTitle(courseID));
+    }
+
+    public String getExamTitle(Long examID) {
+        return titleMap.computeIfAbsent("exam" + examID, ignored -> examRepository.getExamTitle(examID));
+    }
+
+    public String getExerciseTitle(Long exerciseId) {
+        return titleMap.computeIfAbsent("exercise" + exerciseId, ignored -> exerciseRepository.getExerciseTitle(exerciseId));
+    }
+
+    public String getLectureTitle(Long lectureId) {
+        return titleMap.computeIfAbsent("lecture" + lectureId, ignored -> lectureRepository.getLectureTitle(lectureId));
+    }
+
+    public String getOrganizationTitle(Long organizationId) {
+        return titleMap.computeIfAbsent("organization" + organizationId, ignored -> organizationRepository.getOrganizationTitle(organizationId));
+    }
+
+    public String getHintTitle(Long hintId, Long exerciseId) {
+        return titleMap.computeIfAbsent("hint" + hintId + "-" + exerciseId, ignored -> exerciseHintRepository.getHintTitle(hintId));
+    }
+
+    public void setDiagramTitle(Long diagramID, String title) {
+        titleMap.put("diagram" + diagramID, title);
+    }
+
+    public void setCourseTitle(Long courseID, String title) {
+        titleMap.put("course" + courseID, title);
+    }
+
+    public void setExamTitle(Long examID, String title) {
+        titleMap.put("exam" + examID, title);
+    }
+
+    public void setExerciseTitle(Long exerciseId, String title) {
+        titleMap.put("exercise" + exerciseId, title);
+    }
+
+    public void setLectureTitle(Long lectureId, String title) {
+        titleMap.put("lecture" + lectureId, title);
+    }
+
+    public void setOrganizationTitle(Long organizationId, String title) {
+        titleMap.put("organization" + organizationId, title);
+    }
+
+    public void setHintTitle(Long hintId, Long exerciseId, String title) {
+        titleMap.put("hint" + hintId + "-" + exerciseId, title);
+    }
+
+    /**
+     * Configures Hazelcast for the EntityTitleCache before the HazelcastInstance is created.
+     *
+     * @param config the {@link Config} the EntityTitleCache-specific configuration should be added to
+     */
+    public static void configureHazelcast(Config config) {
+        EvictionConfig evictionConfig = new EvictionConfig() //
+                .setEvictionPolicy(EvictionPolicy.NONE);
+        NearCacheConfig nearCacheConfig = new NearCacheConfig() //
+                .setName(HAZELCAST_TITLE_CACHE + "-local") //
+                .setInMemoryFormat(InMemoryFormat.OBJECT) //
+                .setSerializeKeys(true) //
+                .setInvalidateOnChange(true) //
+                .setTimeToLiveSeconds(0) //
+                .setMaxIdleSeconds(0) //
+                .setEvictionConfig(evictionConfig) //
+                .setCacheLocalEntries(true);
+        config.getMapConfig(HAZELCAST_TITLE_CACHE).setNearCacheConfig(nearCacheConfig);
```

Comment:
Please wrap those lines in ` // @formatter:off` and `// @formatter:on` and remove those slashes :+1:

Reply:
See e8b34e5 :)